### PR TITLE
AMR preparation, part II: add `MeshAdaptor` class

### PR DESCRIPTION
**Pull request by tamiko**
This brings back the old infrastructure that we had so far which supported global refinement.

### CODE REVIEW

File: source/mesh_adaptor.h [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+namespace ryujin
+{
+  /**
+   * Controls the adaptation strategy used in MeshAdaptor.
+   *
+   * @ingroup Mesh
+   */
+  enum class AdaptationStrategy {
+    /**
+     * Performs a simply global refinement at specified timepoints.
```
Comment (ejtovar) [inline source/mesh_adaptor.h]: Performs a _simple_ ...

File: source/mesh_adaptor.h [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+     */
+    //@{
+
+    const MPI_Comm &mpi_communicator_;
+
+    dealii::SmartPointer<const OfflineData<dim, Number>> offline_data_;
+    dealii::SmartPointer<const HyperbolicSystem> hyperbolic_system_;
+    dealii::SmartPointer<const ParabolicSystem> parabolic_system_;
+
+    mutable bool need_mesh_adaptation_;
```
Comment (ejtovar) [inline source/mesh_adaptor.h]: This should not be `mutable`?

**Comment by ejtovar** [comment]
@harmonj FYI.

**Review by ejtovar — DISMISSED** [review]
@tamiko Looks good, just minor changes!

**Comment by tamiko** [comment]
@ejtovar I have addressed your comments.

**Comment by ejtovar** [comment]
@tamiko Thank you! Enabled auto merge.

**Pull request merged**